### PR TITLE
Add fixes for ```dmin1```, ```dmax1``` implementation

### PR DESCRIPTION
**Pull request by Pranavchiku**
With this PR, I am able to succesfully compile ```/scipy/scipy/special/cdflib/bcorr.f``` to ASR
>ASR
```
(TranslationUnit (SymbolTable 1 {bcorr: (Function (SymbolTable 2 {a: (Variable 2 a Local () () Default (Real 8 []) Source Public Required .false.), a0: (Variable 2 a0 Unspecified () () Default (Real 8 []) Source Public Required .false.), b: (Variable 2 b Local () () Default (Real 8 []) Source Public Required .false.), b0: (Variable 2 b0 Unspecified () () Default (Real 8 []) Source Public Required .false.), bcorr: (Variable 2 bcorr ReturnVar () () Default (Real 8 []) Source Public Required .false.), c: (Variable 2 c Local () () Default (Real 8 []) Source Public Required .false.), c0: (Variable 2 c0 Local () (RealConstant 0.833333 (Real 8 [])) Default (Real 8 []) Source Public Required .false.), c1: (Variable 2 c1 Local () (RealUnaryMinus (RealConstant 0.277778 (Real 8 [])) (Real 8 []) (RealConstant -0.277778 (Real 8 []))) Default (Real 8 []) Source Public Required .false.), c2: (Variable 2 c2 Local () (RealConstant 0.793651 (Real 8 [])) Default (Real 8 []) Source Public Required .false.), c3: (Variable 2 c3 Local () (RealUnaryMinus (RealConstant 0.595203 (Real 8 [])) (Real 8 []) (RealConstant -0.595203 (Real 8 []))) Default (Real 8 []) Source Public Required .false.), c4: (Variable 2 c4 Local () (RealConstant 0.837308 (Real 8 [])) Default (Real 8 []) Source Public Required .false.), c5: (Variable 2 c5 Local () (RealUnaryMinus (RealConstant 0.165323 (Real 8 [])) (Real 8 []) (RealConstant -0.165323 (Real 8 []))) Default (Real 8 []) Source Public Required .false.), dmax: (ExternalSymbol 2 dmax 4 dmax lfortran_intrinsic_math2 [] dmax Private), dmin: (ExternalSymbol 2 dmin 4 dmin lfortran_intrinsic_math2 [] dmin Private), h: (Variable 2 h Local () () Default (Real 8 []) Source Public Required .false.), s11: (Variable 2 s11 Local () () Default (Real 8 []) Source Public Required .false.), s3: (Variable 2 s3 Local () () Default (Real 8 []) Source Public Required .false.), s5: (Variable 2 s5 Local () () Default (Real 8 []) Source Public Required .false.), s7: (Variable 2 s7 Local () () Default (Real 8 []) Source Public Required .false.), s9: (Variable 2 s9 Local () () Default (Real 8 []) Source Public Required .false.), t: (Variable 2 t Local () () Default (Real 8 []) Source Public Required .false.), w: (Variable 2 w Local () () Default (Real 8 []) Source Public Required .false.), x: (Variable 2 x Local () () Default (Real 8 []) Source Public Required .false.), x2: (Variable 2 x2 Local () () Default (Real 8 []) Source Public Required .false.)}) bcorr [(Var 2 a0) (Var 2 b0)] [] [(= (Var 2 a) (FunctionCall 2 dmin () [((Var 2 a0)) ((Var 2 b0))] (Real 8 []) () ()) ()) (= (Var 2 b) (FunctionCall 2 dmax () [((Var 2 a0)) ((Var 2 b0))] (Real 8 []) () ()) ()) (= (Var 2 h) (RealBinOp (Var 2 a) Div (Var 2 b) (Real 8 []) ()) ()) (= (Var 2 c) (RealBinOp (Var 2 h) Div (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (Var 2 h) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 x) (RealBinOp (RealConstant 1.000000 (Real 8 [])) Div (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (Var 2 h) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 x2) (RealBinOp (Var 2 x) Mul (Var 2 x) (Real 8 []) ()) ()) (= (Var 2 s3) (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (RealBinOp (Var 2 x) Add (Var 2 x2) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 s5) (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (RealBinOp (Var 2 x) Add (RealBinOp (Var 2 x2) Mul (Var 2 s3) (Real 8 []) ()) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 s7) (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (RealBinOp (Var 2 x) Add (RealBinOp (Var 2 x2) Mul (Var 2 s5) (Real 8 []) ()) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 s9) (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (RealBinOp (Var 2 x) Add (RealBinOp (Var 2 x2) Mul (Var 2 s7) (Real 8 []) ()) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 s11) (RealBinOp (RealConstant 1.000000 (Real 8 [])) Add (RealBinOp (Var 2 x) Add (RealBinOp (Var 2 x2) Mul (Var 2 s9) (Real 8 []) ()) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 t) (RealBinOp (RealBinOp (RealConstant 1.000000 (Real 8 [])) Div (Var 2 b) (Real 8 []) ()) Pow (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToReal (Real 8 []) (RealConstant 2.000000 (Real 8 []))) (Real 8 []) ()) ()) (= (Var 2 w) (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (Var 2 c5) Mul (Var 2 s11) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (RealBinOp (Var 2 c4) Mul (Var 2 s9) (Real 8 []) ()) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (RealBinOp (Var 2 c3) Mul (Var 2 s7) (Real 8 []) ()) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (RealBinOp (Var 2 c2) Mul (Var 2 s5) (Real 8 []) ()) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (RealBinOp (Var 2 c1) Mul (Var 2 s3) (Real 8 []) ()) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (Var 2 c0) (Real 8 []) ()) ()) (= (Var 2 w) (RealBinOp (Var 2 w) Mul (RealBinOp (Var 2 c) Div (Var 2 b) (Real 8 []) ()) (Real 8 []) ()) ()) (= (Var 2 t) (RealBinOp (RealBinOp (RealConstant 1.000000 (Real 8 [])) Div (Var 2 a) (Real 8 []) ()) Pow (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToReal (Real 8 []) (RealConstant 2.000000 (Real 8 []))) (Real 8 []) ()) ()) (= (Var 2 bcorr) (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (RealBinOp (Var 2 c5) Mul (Var 2 t) (Real 8 []) ()) Add (Var 2 c4) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (Var 2 c3) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (Var 2 c2) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (Var 2 c1) (Real 8 []) ()) Mul (Var 2 t) (Real 8 []) ()) Add (Var 2 c0) (Real 8 []) ()) Div (Var 2 a) (Real 8 []) ()) Add (Var 2 w) (Real 8 []) ()) ()) (Return)] (Var 2 bcorr) Source Public Implementation () .false. .false. .false.), iso_fortran_env: (IntrinsicModule lfortran_intrinsic_iso_fortran_env), lfortran_intrinsic_builtin: (IntrinsicModule lfortran_intrinsic_builtin), lfortran_intrinsic_math2: (IntrinsicModule lfortran_intrinsic_math2), lfortran_intrinsic_math3: (IntrinsicModule lfortran_intrinsic_math3)}) [])
```

### CODE REVIEW

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -2652,7 +2652,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 loc);
         }
         std::string module_name = intrinsic_procedures.get_module(remote_sym, loc);
-
+        if(module_name=="dmin1") module_name="dmin";
+        if(module_name=="dmax1") module_name="dmax";
```
Comment (czgdp1807) [inline src/lfortran/semantics/ast_common_visitor.h:2656]:
```suggestion
        if (module_name == "dmin1" || module_name == "dmax1") {
            module_name.pop_back();
        }
```

File: tests/reference/asr-dmax1-ce914e0.stderr [hunk 2]
```diff
@@ -0,0 +1,5 @@
+semantic error: Symbol 'dmax1' is not a function or an array
+ --> tests/dmax1.f90:3:3
+  |
+3 | s=dmax1(a0,b0)
+  |   ^^^^^^^^^^^^
```
Comment (czgdp1807) [inline tests/reference/asr-dmax1-ce914e0.stderr:5]: I think you forgot to delete this file.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: tests/reference/asr-dmax1-ce914e0.stderr (hunk 2 above)
Comment (Pranavchiku) [reply tests/reference/asr-dmax1-ce914e0.stderr:5]: Why I need to delete this? I guess this changes occurred when I run ```./run_tests.py```

Comment (Pranavchiku) [reply tests/reference/asr-dmax1-ce914e0.stderr]: Fixed this

**Comment by Pranavchiku** [comment]
@certik @czgdp1807 I guess this PR can have final review

**Review by konradha — APPROVED** [review]
This looks good, maybe an additional integration test would be nice.

File: tests/dmin1.f90 [hunk 3]
```diff
@@ -0,0 +1,4 @@
+subroutine dcstep()
+double precision a0,b0,s
+s=dmin1(a0,b0)
+end
```
Comment (czgdp1807) [inline tests/dmin1.f90:4]: Please move this and `tests/dmax1.f90` into https://github.com/lfortran/lfortran/blob/main/integration_tests and register them in https://github.com/lfortran/lfortran/blob/main/integration_tests/CMakeLists.txt. Please update these tests for thorough checking of `dmin1` and `dmax1`. Negative, 0 and Positive reals and their combinations.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: tests/dmin1.f90 (hunk 3 above)
Comment (Smit-create) [reply tests/dmin1.f90]: Done

Comment (Smit-create) [reply tests/dmin1.f90]: Done

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**